lib/usd/translators/meshWriter.cpp

```diff
@@ -41,21 +42,178 @@
 #include <pxr/usd/usdSkel/root.h>
 #include <pxr/usd/usdUtils/pipeline.h>
 
+#include <maya/MFloatArray.h>
+#include <maya/MFnBlendShapeDeformer.h>
 #include <maya/MFnDependencyNode.h>
+#include <maya/MFnGeometryFilter.h>
 #include <maya/MFnMesh.h>
+#include <maya/MFnSkinCluster.h>
+#include <maya/MGlobal.h>
 #include <maya/MIntArray.h>
+#include <maya/MItDependencyGraph.h>
 #include <maya/MPlug.h>
 #include <maya/MPlugArray.h>
 #include <maya/MString.h>
 #include <maya/MStringArray.h>
 #include <maya/MUintArray.h>
 
+#include <cfloat>
 #include <set>
 #include <string>
 #include <vector>
 
 PXR_NAMESPACE_OPEN_SCOPE
 
+MStatus mayaDisableAllBlendShapesForMesh(
+    const MObject& mesh,
+    MObjectArray&  blendShapeNodes,
+    MFloatArray&   envelopes)
+{
+    MStatus status;
+    // NOTE: (yliangsiew) If there's a skinCluster, find that first since that
+    // will be the intermediate to the blendShape node. If not, just search for any
+    // blendshape deformers upstream of the mesh.
+    MObjectArray skinClusters;
+    status = UsdMayaMeshWriteUtils::getSkinClustersUpstreamOfMesh(mesh, skinClusters);
+    CHECK_MSTATUS_AND_RETURN_IT(status);
+    unsigned int numSkinClusters = skinClusters.length();
+    MObjectArray searchObjs;
+    if (numSkinClusters == 0) {
+        searchObjs.append(MObject(mesh));
+    } else {
+        searchObjs = MObjectArray(skinClusters);
+    }
+    for (unsigned int i = 0; i < searchObjs.length(); ++i) {
+        MObject            searchObject = searchObjs[i];
+        MFnGeometryFilter  fnGeoFilter;
+        MItDependencyGraph itDg(
+            searchObject,
+            MFn::kBlendShape,
+            MItDependencyGraph::kUpstream,
+            MItDependencyGraph::kDepthFirst,
+            MItDependencyGraph::kPlugLevel,
+            &status);
+        CHECK_MSTATUS_AND_RETURN_IT(status);
+        MFnDependencyNode fnNode;
+        for (; !itDg.isDone(); itDg.next()) {
+            MObject curBlendShape = itDg.currentItem();
+            TF_VERIFY(curBlendShape.hasFn(MFn::kBlendShape));
+            status = fnGeoFilter.setObject(curBlendShape);
+            printf("Disabling: %s\n", fnGeoFilter.name().asChar());
+            CHECK_MSTATUS_AND_RETURN_IT(status);
+            float curEnvelope = fnGeoFilter.envelope();
+            blendShapeNodes.append(curBlendShape);
+            envelopes.append(curEnvelope);
+            fnGeoFilter.setEnvelope(0.0f);
+        }
+    }
+
+    return MStatus::kSuccess;
+}
+
+MStatus
+mayaEnableGeometryFilterNodesForMesh(const MObjectArray& nodes, const MFloatArray& envelopes)
```

Comment:
This is basically because we disable the blendshape nodes in the first call, and
since we already know which nodes we disabled, we don't have to find them again.
The intention here is to write out the points of the mesh in the "pref" pose.